src/llm.py

```diff
@@ -0,0 +1,317 @@
+import asyncio
+import datetime
+import json
+import random
+import re
+from typing import Dict, List, Optional, Tuple
+
+import openai
+import tiktoken
+from nextcord import Client, Member, Message
+
+import config
+
+
+# Initialize globals
+def initialize(client: Client) -> None:
+    global gpt_lock
+    global context_data
+    global encoding
+    global self_user
+    global banned_word_pattern
+    global word_trigger_pattern
+    global user_trigger_pattern
+    global user_info_map
+
+    openai.api_key = config.openai_api_key
+    # Global lock for message response
+    gpt_lock = asyncio.Lock()
+    # Load manual hidden data
+    try:
+        with open(config.llm_context_file) as f:
+            context_data = json.load(f)
+    except (FileNotFoundError, json.decoder.JSONDecodeError) as e:
+        print(
+            f"ERROR: Issue loading {config.llm_context_file}. GPT capabilities will be disabled.\n{e}"
+        )
+        # Signal context data was not loaded correctly
```

Comment:
This comment seems a bit superfluous with the print statement above. Might just want to axe it?